### ISSUE TITLE
Cannot extract TrueHD with AD3 to TrueHD and AC3

### ISSUE THREAD

**Issue by ted423** [justdan96/tsMuxer#553]
ffmpeg can't decode  turehd with ac3 
`00000.track_4352.thd: Invalid data found when processing input`

**Comment by ted423** [comment]
```
ffprobe.exe D:\IT\Rip\Demux\Thugs\00000.track_4352.thd
ffprobe version n5.0-4-g911d7f167c-20220209 Copyright (c) 2007-2022 the FFmpeg developers
  built with gcc 11.2.0 (crosstool-NG 1.24.0.498_5075e1f)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-w64-mingw32- --arch=x86_64 --target-os=mingw32 --enable-gpl --enable-version3 --disable-debug --disable-w32threads --enable-pthreads --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libvorbis --enable-opencl --disable-libpulse --enable-libvmaf --disable-libxcb --disable-xlib --enable-amf --enable-libaom --enable-avisynth --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libass --enable-libbluray --enable-libmp3lame --enable-libopus --enable-librist --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libmfx --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --enable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --disable-libdrm --disable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags= --extra-libs=-lgomp --extra-version=20220209
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100
00000.track_4352.thd: Invalid data found when processing input
```

**Comment by ValZapod** [comment]
This is a known bug. https://trac.ffmpeg.org/ticket/667

**Comment by ted423** [comment]
> This is a known bug. https://trac.ffmpeg.org/ticket/667

so I just want a split file

**Comment by ValZapod** [comment]
> so I just want a split file

Use ffmpeg for that ffmpeg -i file.ts -c copy -vn -sn -map 0:a:0 mfile.mp4 (or thd or ts)

**Comment by ted423** [comment]
> > so I just want a split file
> 
> Use ffmpeg for that `ffmpeg -i file.ts -c copy -vn -sn -map 0:a:0 mfile.mp4 (or thd or ts)`

I need time to do some testing and come up with a concrete response.  Most of the time I read mpls directly.  In some cases,  will need merge multiple m2ts, eac3to and mkvtoolnix will cause some errors

**Comment by ValZapod** [comment]
> merge multiple m2ts,

Segmented m2ts has some segments in TrueHD that should be removed anyway. That can be done with https://github.com/domyd/mlp

**Comment by ted423** [comment]
> > merge multiple m2ts,
> 
> Segmented m2ts has some segments in TrueHD that should be removed anyway. That can be done with https://github.com/domyd/mlp

thanks a lot, I will try kater

**Comment by jcdr428** [comment]
@ValZapod double major sync units should be found only after last access units, which is not the case anymore with domyd. tsMuxer THD might be 100ms out of sync in very, very rare cases similar to Monsters University containing 100+ segments, but at least the THD stream is DD compliant.

**Comment by ValZapod** [comment]
> double major sync units should be found only after last access units

And only once in the very start (spec says it shouldn't really be allowed to have two Majors after each other, other than at the start of the thd file). Recently ffmpeg added keyframes for audio and then added keyframes for truehd. Tsmuxer is completly broken in that sense. Re: DGDemux development

**Comment by jcdr428** [comment]
@ValZapod can you please open a new issue and clarify/detail what is broken/what needs to be fixed ?

**Comment by justdan96** [comment]
It isn't clear what the issue with tsMuxer is - can someone please clarify?

**Comment by ted423** [comment]
> It isn't clear what the issue with tsMuxer is - can someone please clarify?

ffmpeg can't decode turehd with ac3.
So I want tsMuxer can extract it to two file.
ValZapod Try to give me some other solutions.

but some BD has multi m2ts, useac3to and mkvtoolnix make some sync issue like this https://github.com/justdan96/tsMuxer/issues/262

I have not tested tsMuxer on Truehd whether there are similar problems, and these discussions are out of topic

**Comment by ValZapod** [comment]
>ffmpeg can't decode turehd with ac3.

Yes, that is a new bug. See https://trac.ffmpeg.org/ticket/667#comment:4 (before you could use -f ac3 to force ac3 only out of ac3+thd interleave). Wow :(
